### PR TITLE
azureclient track2:customize default transporter in track2 client

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
customize default transport for all of track2 clients
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azureclients/v2/utils/options.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ func GetDefaultOption(apiVersion string) *arm.ClientOptions {
 				custompolicy.NewThrottlingPolicy(),
 			},
 			APIVersion: apiVersion,
+			Transport:  defaultHTTPClient,
```
Comment (feiskyer) [inline pkg/azureclients/v2/utils/options.go:40]: does it work with https://github.com/Azure/go-armbalancer?

Comment (MartinForReal) [reply pkg/azureclients/v2/utils/options.go:40]: added go-armbalancer.

**Review by feiskyer — APPROVED** [review]
LGTM, thanks for optimizing the default transport

**Pull request merged**